sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PruneFileSourcePartitions.scala

```diff
@@ -59,6 +60,21 @@ private[sql] object PruneFileSourcePartitions extends Rule[LogicalPlan] {
     Project(projects, withFilter)
   }
 
+  private def getNewStatistics(
+              fileIndex: InMemoryFileIndex,
```

Comment:
> @fuwhu `InMemoryFileIndex` caches all the file status on construction. Is it true that statistic calculation is very expensive?

yea, you are right. The refresh0 method will call listLeafFiles to get all file status, which already implement listing in parallel when path number exceed threshold.
So the statistic calculation here is actually just a sum of the length of all leaf files. Will remove the conf check here.

But for hive table, currently, there is no some place to get the file/dir status directly without accessing the file system. So i think it is still necessary to limit the partition number of statistic calculation. WDYT ?

For parallel statistic calculation, i am not sure whether it is worthwhile to start a distributed job to do the statistic calculation or start multiple threads to do it ? The benefit of statistic computation may not cover the cost of the statistic calculation. WDYT ?

Reply:
I don't think it is a good idea to limit the number of partitions.

Reply:
So you prefer to do the statistic calculation in parallel ?
@cloud-fan WDYT?

Reply:
The file listing is parallel, and the statistic calculation is just getting the sum of the files. So no need to make the statistic calculation parallel. 
I updated my comments minutes after I left them.

Reply:
> the statistic calculation is just getting the sum of the files

this is true in PruneFileSourcePartitions, because InMemoryFileIndex already did file listing. 
I already partition number check in PruneFileSourcePartitions.
but this is not true in PruneHiveTablePartitions, the size of each hive table partition need to be calculated via HDFS if it's not available in meta data.

Please correct me if i am wrong, thanks.